BasicsOOP/Program.cs

```diff
@@ -1,5 +1,225 @@
+    var numb2 = new ComplexNumber(-5, 3);
+
+    Console.WriteLine("\nТест команды +");
+    numb1.Real = 4; numb1.Imaginary = 2;
+    numb2.Real = -5; numb2.Imaginary = 3;
+    Console.WriteLine(numb1 + " + " + numb2 + " = " + (numb1+ numb2));
+    Console.Write("Совпадает ли с ожидаемым результатом: ");
+    Console.WriteLine(numb1 + numb2 == new ComplexNumber(-1, 5));
+
+    Console.WriteLine("\nТест команды *");
+    numb1.Real = 2; numb1.Imaginary = 3;
+    numb2.Real = 5; numb2.Imaginary = 4;
+    Console.WriteLine(numb1 + " * " + numb2 + " = " + (numb1 * numb2));
+    Console.Write("Совпадает ли с ожидаемым результатом: ");
+    Console.WriteLine(numb1 * numb2 == new ComplexNumber(-2, 23));
+
+    Console.WriteLine("\nТест команды -");
+    numb1.Real = 5; numb1.Imaginary = 4;
+    numb2.Real = -2; numb2.Imaginary = 3;
+    Console.WriteLine(numb1 + " - " + numb2 + " = " + (numb1 - numb2));
+    Console.Write("Совпадает ли с ожидаемым результатом: ");
+    Console.WriteLine(numb1 - numb2 == new ComplexNumber(7, 1));
+
+    Console.WriteLine("\nТест команды ==");
+    numb1.Real = 5; numb1.Imaginary = 4;
+    numb2.Real = -2; numb2.Imaginary = 3;
+    Console.WriteLine(numb1 + " == " + numb2 + " = " + (numb1 == numb2));
+    Console.Write("Совпадает ли с ожидаемым результатом: ");
+    Console.WriteLine((numb1 == numb2) == false);
+
+    numb1.Real = -2; numb1.Imaginary = 3;
+    numb2.Real = -2; numb2.Imaginary = 3;
+    Console.WriteLine(numb1 + " == " + numb2 + " = " + (numb1 == numb2));
+    Console.Write("Совпадает ли с ожидаемым результатом: ");
+    Console.WriteLine((numb1 == numb2) == true);
+}
+
+static void SetDefault(ref RationalNumber numb23, ref RationalNumber numb58_1, ref RationalNumber numb58_2, ref RationalNumber numb21_18)
+{
+    numb23.Numerator = 2;
+    numb23.Denominator = 3;
+
+    numb58_1.Numerator = 5;
+    numb58_1.Denominator = 8;
+
+    numb58_2.Numerator = 5;
+    numb58_2.Denominator = 8;
+
+    numb21_18.Numerator = 21;
+    numb21_18.Denominator = 18;
+}
+//SetDefault(ref numb23, ref numb58_1, ref numb58_2, ref numb21_18);
+static void TestRationalNumber()
+{
+    var numb23 = new RationalNumber(2, 3); //0,66(..67)
+    var numb58_1 = new RationalNumber(5, 8); //0,625
+    var numb58_2 = new RationalNumber(5, 8); //0,625
+    var numb21_18 = new RationalNumber(21, 18); //1,166(..67)
+    // 21/18 > 2/3 > 5/8
+    var tmp = new RationalNumber();
+
+
+    //Тест в виде: Выводит на экран всегда true, если результат совпадает с ожидаемым
+    //Команда в формате: ВыводНаЭкран(func() == ожидаемый_результат);
+
+    try
+    {
+
+        //==
+        Console.WriteLine("\nТест команды ==");
+        Console.WriteLine((numb58_1 == numb23) == false);
+        Console.WriteLine((numb58_1 == numb58_2) == true);
+        Console.WriteLine((numb23 == numb21_18) == false);
+
+        //!=
+        Console.WriteLine("\nТест команды !=");
+        Console.WriteLine((numb58_1 != numb23) == true);
```

Comment:
Индусский код. То, что написано в скобках - это уже булево значение.
```suggestion
        Console.WriteLine(numb58_1 != numb23);
```

Reply:
Выше и ниже - тоже самое.